client/src/charts/nivo/NivoCharts.js

```diff
@@ -694,21 +695,27 @@ export class NivoResponsiveLine extends React.Component {
         .fromPairs()
         .assign({label: row.id})
         .value(),
-      sort_values: {
-        label: row.id,
-        ...(_.omit(row, 'id')),
-      },
+      sort_values: _.chain(row.data)
+        .map(d => [d.x,d.y])
+        .fromPairs()
+        .assign({label: row.id})
+        .value(),
       search_values: {
         label: row.id,
       },
     }) );
-    const ordered_column_keys = _.concat(['label'], _.chain(data)
+    const last_column_keys = _.chain(data)
       .map( d=>_.map( d.data, d=>d.x ) )
       .flatten()
       .uniq()
-      .value() );
-    const column_names = _.fromPairs(_.zip(ordered_column_keys,ordered_column_keys)); //TODO: fix this
-    
+      .value();
+    const ordered_column_keys = _.concat(['label'], last_column_keys);
+  
+    const column_names = _.chain(ordered_column_keys)
+      .zip( _.concat([table_first_column_name ? table_first_column_name : text_maker("label")], last_column_keys) )
```

Comment:
Super minor nit pick on my personal preference to short circuiting:
`_.concat([table_first_column_name || text_maker("label")], last_column_keys)`

Reply:
:+1: